neqo-transport/src/connection.rs

```diff
@@ -2098,9 +2098,19 @@ impl Connection {
     /// Returns how many bytes were successfully sent. It can only return 0 or exactly amount of data
     /// supply in the buffer.
     pub fn stream_send_atomic(&mut self, stream_id: u64, data: &[u8]) -> Res<usize> {
-        self.send_streams
+        let val = self
+            .send_streams
             .get_mut(stream_id.into())?
-            .send_atomic(data)
+            .send_atomic(data);
+        if let Ok(val) = val {
+            debug_assert!(
+                val == 0 || val == data.len(),
```

Comment:
I would like this to be an error. If this fails we cannot really continue. qpack will be completely out of sync.

Reply:
So should this be an assert instead of debug assert? This should never happen and users of this method should not have to check this themselves.

Reply:
debug_assert is fine.

Reply:
Should send_atomic return bool rather than usize?

Reply:
sure let's try that.